### DISCUSSION TITLE
Cascade REMOVE on primary key raise error: "column "video_id" is in a primary key" ?

### DISCUSSION THREAD

**Discussion by dungkaka** [mikro-orm/mikro-orm#4909] [General]
I have entity:

```ts
class LessonVideo {
  @ManyToOne({ entity: () => Lesson, primary: true })
    lesson: Ref<Lesson>;

 @ManyToOne({ entity: () => Video, cascade: [Cascade.REMOVE], primary: true  })
    video: Ref<Video>;
}
```

Migration throw error: column "video_id" is in a primary key.

I'm not sure I skip some rules on the document ?

**Comment by B4nan** [comment]
Please don't ignore the issue template and always try to provide as many (relevant) details as you can. If you see some error, always provide the full error message as well as the full stack trace. Or just open a discussion instead, I want to keep the issues only for bugs and things that are actionable, not for questions.

Also, do you really want `Cascade.REMOVE`? People often misunderstand what it does, its for app level cascading, maybe you wanted `onDelete: 'cascade'` instead (which only affects the referential integrity). The docs explicitly discourage using `Cascade.REMOVE` on M:1 properties as it can have unwanted side effects:

https://mikro-orm.io/docs/next/cascading#cascade-remove
